agent/proxycfg/testing_tproxy.go

```diff
@@ -128,7 +128,7 @@ func TestConfigSnapshotTransparentProxy(t testing.T) *ConfigSnapshot {
 	})
 }
 
-func TestConfigSnapshotTransparentProxyHTTPUpstream(t testing.T, additionalEntries ...structs.ConfigEntry) *ConfigSnapshot {
+func TestConfigSnapshotTransparentProxyHTTPUpstream(t testing.T, nsFn func(ns *structs.NodeService), additionalEntries ...structs.ConfigEntry) *ConfigSnapshot {
```

Comment:
This allows us to modify the local service defaults to include the Lua extension, which will target TProxy upstreams